### PR TITLE
WIP: Contact form effect validation

### LINKED ISSUE #363: Rebuild the contact form and validation
- Getting status 406 from the server.
- What should be displayed to the user in case of some server error?
- Do we need to return the users to the form with the given data that has been submitted?

### PR DESCRIPTION
**Pull request by DavidBronfen**
Fix #363

### CODE REVIEW

**Comment by DavidBronfen** [comment]
image

**Comment by DavidBronfen** [comment]
image

**Comment by DavidBronfen** [comment]
@nirgn975 Please take a look at the Travis last run, it fails there because of some wrong imports I guess, not sure why is it happen.
image


I and @danielp16 trued to look at the 406 error and we think stems because it tries to send an email and something wrong there, this is the reason for the 406 status I'm getting from the server.

Any sugestion?

File: client/src/app/modules/contact/reducers/contact.reducer.ts [hunk 1]
```diff
@@ -29,6 +29,16 @@ export function reducer(state = initialState, action: ContactActions): State {
...
+        submitionState: true
+      } as Contact;
+    }
+
+    case ContactActionTypes.SEND_NEW_CONTACT_FAILED: {
+      return {
+        name: action.payload.name,
+        email: action.payload.email,
+        content: action.payload.content,
+        submitionState: false
```
Comment (nirgn975) [inline client/src/app/modules/contact/reducers/contact.reducer.ts]: Add comma (`,`) at the end of the line.

File: client/src/app/modules/contact/reducers/contact.reducer.ts [hunk 2]
```diff
@@ -29,6 +29,16 @@ export function reducer(state = initialState, action: ContactActions): State {
         name: action.payload.name,
         email: action.payload.email,
         content: action.payload.content,
+        submitionState: true
```
Comment (nirgn975) [inline client/src/app/modules/contact/reducers/contact.reducer.ts]: Add comma (`,`) at the end of the line.

**Comment by nirgn975** [comment]
@DavidBronfen Also, for it to work you need to put the email password of the openpension gmail account in the `settings.py` file at `server/config`, at line 197.

**Comment by DavidBronfen** [comment]
@nirgn975 Looks like there is an error in the server scripts:
image

Locally all the tests pass and the functionality works correctly.

**Pull request merged**